### DISCUSSION TITLE
Deprecation Warning when using BigqueryHook

### DISCUSSION THREAD

**Discussion by fr-shimeng** [apache/airflow#26426] [Q&A]
Hi I'm using airflow of version 2.2.5 on gcp composer, and I'm trying to use a custom operator
to get results from bigquery with the following codes. 
Everything works fine but I got deprecation warning like this:
Could you help me understand the reason why this warning could happen and if I should change my code, please?
```
{warnings.py:109} WARNING - /opt/python3.8/lib/python3.8/site-packages/airflow/providers/google/cloud/hooks/bigquery.py:131: DeprecationWarning: This method will be deprecated. Please use `BigQueryHook.get_client` method
{credentials_provider.py:324} INFO - Getting connection using `google.auth.default()` since no key file is defined for hook.
 {warnings.py:109} WARNING - /opt/python3.8/lib/python3.8/site-packages/airflow/providers/google/cloud/hooks/bigquery.py:2063: DeprecationWarning: This method is deprecated. Please use `BigQueryHook.insert_job` method.
```
```
from airflow.models.baseoperator import BaseOperator
from airflow.providers.google.cloud.hooks.bigquery import BigQueryHook
from plugins import const
GCP_CONNECTION_ID = const.GCP_CONNECTION_ID

class BQQueryResultsOperator(BaseOperator):
    def __init__(self, sql_path:str, **kwargs) -> None:
        super().__init__(**kwargs)
        self.sql_path = sql_path

    def execute(self, context):
        hook = BigQueryHook(
            gcp_conn_id=GCP_CONNECTION_ID,
            location='asia-northeast1',
            use_legacy_sql=False
        )
        with open(self.sql_path, 'r') as f:
            sql = f.read()
            result = hook.get_records(sql)
            return(result)
```

**Comment by potiuk** [comment]
The best is to look at the code of the bigquery Hook where it is in question https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/hooks/bigquery.py and you should also take a look at examples and follow them https://airflow.apache.org/docs/apache-airflow-providers-google/stable/example-dags.html (appropriate for your version of google provider). Also I would strongly recommend to upgrade to latest version of Airflow (and providers) as they have many errors fixed and likely better error messages.
